tun/tun_illumos.go

```diff
@@ -0,0 +1,271 @@
+/* SPDX-License-Identifier: MIT
+ *
+ * Copyright 2019 Joyent, Inc.
+ */
+
+// +build illumos,amd64
+
+package tun
+
+import (
+	"fmt"
+
+	"golang.org/x/sys/unix"
+	"os"
+)
+
+// NativeTun is the OS-specific implementation of a tun interface.
+type NativeTun struct {
+	tunFile *os.File /* TUN device file */
```

Comment:
These comments, `/* TUN device file */` are obvious enough that they probably aren't necessary. If you do think there's something non-obvious going on, use `//` comments for single-line struct member annotation.